### PR TITLE
MSEARHC-45: Implement DELETE/GET /_/tenant endpoint

### PR DESCRIPTION
**Pull request by bohdan-suprun**
## Changes:
* Expose `DELETE/GET` `/_/tenant` endpoints;
* Remove elasticsearch index on `DELETE` `/_/tenant`;
* Use Posrgres testcontainer instead of H2 because base DELETE tenant uses Postgres specific features;
* Move postgres/kafka/elasticsearch container management to extensions, so that they can be reused in other component tests.

### CODE REVIEW

File: src/test/java/org/folio/search/service/IndexServiceTest.java [hunk 1]
```diff
@@ -111,4 +111,24 @@ void reindexInventory_positive() {
 
     verify(instanceStorageClient).submitReindex();
   }
+
+  @Test
+  void shouldDropIndexWhenExists() {
+    var index = "instance_diku";
```
Comment (pfilippovEPAM) [inline src/test/java/org/folio/search/service/IndexServiceTest.java:117]: maybe use existing constant INDEX_NAME from TestConstants class?

File: src/test/java/org/folio/search/support/base/BaseIntegrationTest.java [hunk 2]
```diff
@@ -40,62 +35,36 @@
...
 import org.springframework.test.web.servlet.ResultActions;
-import org.testcontainers.containers.GenericContainer;
-import org.testcontainers.containers.KafkaContainer;
-import org.testcontainers.images.builder.ImageFromDockerfile;
-import org.testcontainers.utility.DockerImageName;
 
 @Log4j2
 @SpringBootTest
 @AutoConfigureMockMvc
+@EnablePostgres
```
Comment (pfilippovEPAM) [inline src/test/java/org/folio/search/support/base/BaseIntegrationTest.java:45]: nice solution :thumbsup:

**Review by pfilippovEPAM — APPROVED** [review]

**Review by hjiebsco — APPROVED** [review]

**Comment by hjiebsco** [comment]
Noticed that tenant API is still at 1.2 not 2.0 yet.

**Comment by bohdan-suprun** [comment]
> Noticed that tenant API is still at 1.2 not 2.0 yet.

Yes, the folio-spring-base implements only this version.

I thought the 1.2 is still supported with 2.0, right?

**Comment by hjiebsco** [comment]
> > Noticed that tenant API is still at 1.2 not 2.0 yet.
> 
> Yes, the folio-spring-base implements only this version.
> 
> I thought the 1.2 is still supported with 2.0, right?

That's right @bohdan-suprun. I was just wondering since other modules have moved to 2.0.

**Comment by bohdan-suprun** [comment]
> That's right @bohdan-suprun. I was just wondering since other modules have moved to 2.0.

I think most of non-RMB modules are still with 1.2 version. Probably we need a story for folio-spring-base to implement it for all of the modules. What do you think?

**Comment by hjiebsco** [comment]
> > That's right @bohdan-suprun. I was just wondering since other modules have moved to 2.0.
> 
> I think most of non-RMB modules are still with 1.2 version. Probably we need a story for folio-spring-base to implement it for all of the modules. What do you think?

Makes sense. Keep modules consistent would be nice.

**Comment by bohdan-suprun** [comment]
@hjiebsco, https://issues.folio.org/browse/FOLSPRINGB-17 created.

File: src/test/java/org/folio/search/service/IndexServiceTest.java (hunk 1 above)
Comment (bohdan-suprun) [reply src/test/java/org/folio/search/service/IndexServiceTest.java]: Thanks, changed.

**Pull request merged**